index.ts

```diff
@@ -70,20 +70,33 @@ class SopsSecretsManagerProvider extends cdk.Construct {
 }
 
 export class SopsSecretsManager extends cdk.Construct {
-    public readonly secret: secretsManager.ISecret;
+    public readonly secret: secretsManager.Secret | undefined;
+    public readonly secretArn: string;
     public readonly asset: s3Assets.Asset;
 
     constructor(scope: cdk.Construct, id: string, props: SopsSecretsManagerProps) {
         super(scope, id);
 
-        this.secret = this.getSecret(props.secret, props.secretName);
+        if(props.secret && props.secretName) {
+            throw new Error("Cannot set both secret and secretName")
+        } else if(props.secret) {
+            this.secretArn = props.secret.secretArn;
+            this.secret = undefined;
```

Comment:
This line feels really weird, but I think I'm persuaded that it makes sense.

If you've passed in the `Secret` or `ISecret`, then you don't need it returned back to you, I guess.